c/tskit/tables.c

```diff
@@ -6719,6 +6724,10 @@ tsk_table_collection_check_mutation_integrity(
                 }
             }
 
+            if (j > 0 && mutations.site[j - 1] != mutations.site[j]) {
```

Comment:
I'm nitpicking here, but we could avoid the j > 0 comparison and be a bit more cache friendly if we put this check into the (j > 0) block around 6706. It probably doesn't matter, but we do run this every time we create a tree sequence object, so it makes sense to keep it as tight as we can.;

Reply:
Nah, not a nickpit! Moved in https://github.com/tskit-dev/tskit/pull/726/commits/79dd9ecd10227631a7dec75c722fb109bb98cc9b